fsspec/implementations/hdfs.py

```diff
@@ -42,21 +118,91 @@ def __init__(
         extra_conf: None or dict
             Passed on to HadoopFileSystem
         """
-        if self._cached:
-            return
-        AbstractFileSystem.__init__(self, **kwargs)
-        self.pars = (host, port, user, kerb_ticket, driver, extra_conf)
-        pahdfs = HadoopFileSystem(
+        super().__init__(**kwargs)
+
+        self.client = HadoopFileSystem(
             host=host,
             port=port,
             user=user,
             kerb_ticket=kerb_ticket,
             driver=driver,
             extra_conf=extra_conf,
         )
-        weakref.finalize(self, lambda: pahdfs.close())
-        self.pahdfs = pahdfs
+        weakref.finalize(self, lambda: self.client.close())
+
+        self.pars = (host, port, user, kerb_ticket, driver, extra_conf)
+
+    @staticmethod
+    def _get_kwargs_from_urls(path):
+        ops = infer_storage_options(path)
+        out = {}
+        if ops.get("host", None):
+            out["host"] = ops["host"]
+        if ops.get("username", None):
+            out["user"] = ops["username"]
+        if ops.get("port", None):
+            out["port"] = ops["port"]
+        return out
+
+    @classmethod
+    def _strip_protocol(cls, path):
+        ops = infer_storage_options(path)
+        return ops["path"]
+
+    def __reduce_ex__(self, protocol):
+        return PyArrowHDFS, self.pars
+
+    def close(self):
+        self.client.close()
 
+    @wrap_exceptions
+    def ls(self, path, detail=True):
+        listing = [
+            self._adjust_entry(entry) for entry in self.client.ls(path, detail=True)
+        ]
+
+        if detail:
+            return listing
+        else:
+            return [entry["name"] for entry in listing]
+
+    @wrap_exceptions
+    def info(self, path):
+        return self._adjust_entry(self.client.info(path))
+
+    def _adjust_entry(self, original_entry):
+        entry = original_entry.copy()
+        if "type" not in entry:
+            entry["type"] = entry["kind"]
+        if "name" not in entry:
+            entry["name"] = self._strip_protocol(entry["path"])
+        return entry
+
+    @wrap_exceptions
+    def cp_file(self, lpath, rpath, **kwargs):
+        with self.open(lpath) as lstream:
+            tmp_fname = "/".join([self._parent(rpath), f".tmp.{secrets.token_hex(16)}"])
+            try:
+                with self.open(tmp_fname, "wb") as rstream:
+                    shutil.copyfileobj(lstream, rstream)
+                self.client.mv(tmp_fname, rpath)
+            except BaseException:  # noqa
```

Comment:
This seems odd to me. I would at least document the process in the method docstring.